presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterConfig.java

```diff
@@ -20,94 +20,96 @@
 @SuppressWarnings("unused")
 public class OrcFileWriterConfig
 {
-    private OrcWriterOptions options = new OrcWriterOptions();
+    private final OrcWriterOptions.Builder optionsBuilder = OrcWriterOptions.builder();
 
-    public OrcWriterOptions toOrcWriterOptions()
+    public OrcWriterOptions.Builder toOrcWriterOptionsBuilder()
     {
-        return options;
+        // toOrcWriterOptions could be called concurrently from multiple threads.
+        // Copy the builder for isolation and  thread safety.
```

Comment:
1) Fix the double space before `thread safety.`
2) I'd rephrase  this in terms of immutability, and drop everything else about isolation and threads.

Reply:
In my understanding using a copy is for isolation and less about immutability. Each user gets its own copy, so it has to do with isolation.

Reply:
Isolation and thread safety are side effects of immutability.